### PR TITLE
arrow: Update to 10.0.0

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

File: mingw-w64-arrow/PKGBUILD [hunk 1]
```diff
@@ -40,13 +39,14 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-boost"
              "${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-rapidjson")
 options=("staticlibs" "strip" "!buildflags")
-source=("https://archive.apache.org/dist/${_realname}/${_realname}-${pkgver}/apache-${_realname}-${pkgver}.tar.gz"{,.asc}
+source=("apache-arrow-${pkgver}.tar.gz"::"https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz"
+        "apache-arrow-${pkgver}.tar.gz.asc"::"https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz.asc"
```
Comment (MehdiChinoune) [inline mingw-w64-arrow/PKGBUILD:43]: Is there a reason to change download link?

**Comment by MehdiChinoune** [comment]
gdal also needs to be rebuilt.

File: mingw-w64-arrow/PKGBUILD (hunk 1 above)
Comment (kou) [reply mingw-w64-arrow/PKGBUILD:43]: This is the official download URL.

See also 1: https://infra.apache.org/release-download-pages.html#download-scripts

See also 2: https://archive.apache.org/dist/

> Please do not download from apache.org! If you are currently at apache.org and would like to browse, please visit [a nearby mirror site](https://www.apache.org/dyn/closer.cgi) instead.

**Comment by kou** [comment]
> gdal also needs to be rebuilt.

Done.

**Pull request merged**